12-make_cov.py

```diff
@@ -0,0 +1,125 @@
+
+    if config.use_ica or config.use_ssp:
+        extension = '_cleaned-epo'
+    else:
+        extension = '-epo'
+
+    epo_fname = op.join(deriv_path, bids_basename + '%s.fif' % extension)
+    cov_fname = op.join(deriv_path, bids_basename + '-cov.fif')
+
+    msg = (f"Computing regularized covariance based on epochs' baseline "
+           f"periods. Input: {epo_fname}, Output: {cov_fname}")
+    logger.info(gen_log_message(message=msg, step=11, subject=subject,
+                                session=session))
+
+    epochs = mne.read_epochs(epo_fname, preload=True)
+
+    # Do not shuffle the data before splitting into train and test samples.
+    # Perform a block cross-validation instead to maintain autocorrelated
+    # noise.
+    cv = KFold(3, shuffle=False)
+    cov = mne.compute_covariance(epochs, tmin=tmin, tmax=tmax, method='shrunk',
+                                 cv=cv, rank='info')
+    cov.save(cov_fname)
+
+
+def compute_cov_from_empty_room(subject, session):
+    deriv_path = config.get_subject_deriv_path(subject=subject,
+                                               session=session,
+                                               kind=config.get_kind())
+
+    bids_basename = make_bids_basename(subject=subject,
+                                       session=session,
+                                       task=config.get_task(),
+                                       acquisition=config.acq,
+                                       run=None,
+                                       processing=config.proc,
+                                       recording=config.rec,
+                                       space=config.space)
+
+    raw_er_fname = op.join(deriv_path,
+                           bids_basename + '_emptyroom_filt_raw.fif')
+    cov_fname = op.join(deriv_path, bids_basename + '-cov.fif')
+
+    extra_params = dict()
+    if not config.use_maxwell_filter and config.allow_maxshield:
+        extra_params['allow_maxshield'] = config.allow_maxshield
+
+    msg = (f'Computing regularized covariance based on empty-room recording. '
+           f'Input: {raw_er_fname}, Output: {cov_fname}')
+    logger.info(gen_log_message(message=msg, step=11, subject=subject,
+                                session=session))
+
+    raw_er = mne.io.read_raw_fif(raw_er_fname, preload=True, **extra_params)
+
+    # Do not shuffle the data before splitting into train and test samples.
+    # Perform a block cross-validation instead to maintain autocorrelated
+    # noise.
+    cv = KFold(3, shuffle=False)
+    cov = mne.compute_raw_covariance(raw_er, method='shrunk', cv=cv,
+                                     rank='info')
+    cov.save(cov_fname)
+
+
+@failsafe_run(on_error=on_error)
+def run_covariance(subject, session=None):
+    if config.meg_noise_cov == 'emptyroom' and 'eeg' not in config.ch_types:
+        compute_cov_from_empty_room(subject=subject, session=session)
+    else:
+        tmin, tmax = config.meg_noise_cov
```

Comment:
I would rename meg_noise_cov to noise_cov as you see here that the tmin, tmax can also be used for eeg data.

Reply:
Done in d918af8